### DISCUSSION TITLE
Import: Metadata is swapped

### DISCUSSION THREAD

**Discussion by azukaar** [photoprism/photoprism#1903] [Q&A]
I recently imported 9k photos from my Iphone. One thing that I noticed is that the metadata has been completely wrecked. Here is an example of a picture from my Iphone:

![Capture](https://user-images.githubusercontent.com/7872597/148541559-264c90ba-ecd9-4919-9ff2-fca112e723e5.JPG)

On this capture, you can see the metadata set this photo to have been taken in NYC in 2018. The thing is, this photo was actually taken in London in 2020. Here's a screenshot in my Iphone showing the proper date:

![image](https://user-images.githubusercontent.com/7872597/148543025-b6d43fbb-ba20-4f71-b9f1-c26ddffb4a54.png)

Quite random! But not so much, here's the interesting bit: in 2018, I was indeed living in NYC where I took a fair amount of picture on that same phone. When I go through my pictures in Photoprism, I can see many London pictures messed up in New York, 2-3 years earlier. Which leads me to believe that somehow those pictures have been tagged based on other pictures' metadata.

I can find a fair amount of London 2020-2021 pictures messed up as NYC 2018-2019 pictures, but not the other way around if that matter. Also I attached a picture, downloaded from photoprism directly, where you can clearly see that the picture is named 2018XXXX by photoprism's import when the metadata still says 2020.

![image](https://user-images.githubusercontent.com/7872597/148542424-5971ebf1-2708-4a1f-a383-4759ce8c4d64.png)

[20180909_181828_5BFD9DA9.zip](https://github.com/photoprism/photoprism/files/7828714/20180909_181828_5BFD9DA9.zip)

I am using Photoprism 211215-93b26f19-Linux-x86_64
I did the import with PhotoSync as recommended by te doc.

**Comment by lastzero** [comment]
From what I see, it's a stack with TWO files. So the metadata may be from the other file?

**Comment by lastzero** [comment]
### For what reasons can files be stacked? ###

1. Files sharing exactly the same file and folder name will always be stacked, for example `/2018/IMG_1234.jpg` and `/2018/IMG_1234.avi`
2. Files with related, sequential names like `/2018/IMG_1234 (2).jpg` and `/2018/IMG_1234 (3).jpg` may be stacked as well (optional)
3. Pictures were taken at the same GPS position and second (optional)
4. Image metadata contains the same *Unique Image ID* or *XMP Instance ID* (optional)

You can change the behaviour for 2 - 4 in [*Settings*](https://docs.photoprism.app/user-guide/settings/library/).

**Comment by azukaar** [comment]
Thanks for your answer!
It is stacked indeed, because it is from an Iphone, the other file is the HEIC file (actually if I understand, I uploaded the HEIC and Photoprism generated the JPEG?).
Good find though, since the other image, the original HEIC, is actually not the same picture as what the JPEG is!

![image](https://user-images.githubusercontent.com/7872597/148569501-c697ee03-05d2-4846-9b60-3a8314fcd833.png)

Is Photoprism generating those?

**Comment by lastzero** [comment]
If it's not the same content, PhotoPrism didn't create them.

> New sidecar files will be created in the storage folder by default so that the originals folder can be mounted read-only.

see https://docs.photoprism.app/getting-started/faq/

**Comment by lastzero** [comment]
When exporting iPhone images it's a common problem that unrelated files get the same same like in your case. I had similar issues. Ideally, you can change settings or make sure different folders are used for unrelated files.

**Reply by azukaar** [reply]
Guessing you mean same name, that said on the iphone the photos are sorted by year, so 2018 and 2021 would be in two different folders I think (I could see the year subfolders in the import folders), two picture with the same name but in different years should probably not be paired right?

**Reply by lastzero** [reply]
When you export files, they may be grouped / named differently. Thus we recommend PhotoSync as it has an easy UI and you can control file names.

**Reply by azukaar** [reply]
This is exactly from a Photosync import

**Reply by lastzero** [reply]
Did PhotoSync group it wrong then? It's definitely a problem with stacked files, right?

**Reply by azukaar** [reply]
not sure how could i verify whether its photoprism or photosync who grouped the photos wrong tbh
in my import/ folder i have seen the year based folders, so im assuming it kept the folder structure
would it be possible that somehow photoprism ignored thre folder name or something?

**Reply by lastzero** [reply]
No, it should not. Needs further investigation next week. Have a great weekend!

**Reply by azukaar** [reply]
Thanks! you too
